src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Converters/SystemJson/ListJsonConverterTests.cs

```diff
@@ -0,0 +1,374 @@
+#nullable enable
+using System.Collections.Generic;
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Xunit;
+using System.Linq;
+using GovUk.Education.ExploreEducationStatistics.Common.Converters.SystemJson;
+using AngleSharp.Text;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Converters.SystemJson;
+
+public class ListJsonConverterTests
+{
+    private abstract class SampleClass<T>
+    {
+        public abstract IReadOnlyList<T>? SampleField { get; set; }
+    }
+
+    private record SampleObject(string Name);
+
+    private enum SampleEnum
+    {
+        SampleValue1,
+        SampleValue2
+    }
+
+    private class ClassWithObjectConverter : SampleClass<SampleObject>
+    {
+        [JsonConverter(typeof(ListJsonConverter<SampleObject, SampleObjectJsonConverter>))]
+        public override IReadOnlyList<SampleObject>? SampleField { get; set; }
+    }
+
+    private class ClassWithEnumConverter : SampleClass<SampleEnum>
+    {
+        [JsonConverter(typeof(ListJsonConverter<SampleEnum, JsonStringEnumConverter>))]
+        public override IReadOnlyList<SampleEnum>? SampleField { get; set; }
+    }
```

Comment:
Reorganised and removed equal and hash code methods. We can use the `AssertDeepEqualTo` extension to check that objects with lists are equal.